zephyr/common/soc_init.c

```diff
@@ -45,11 +46,12 @@ void print_banner(void)
 int read_bootloader_header(void)
 {
 	/* load bootloader image header */
-	if (bootloader_flash_read(FIXED_PARTITION_OFFSET(boot_partition), &bootloader_image_hdr,
-				  sizeof(esp_image_header_t), true) != 0) {
+	if (bootloader_flash_read_rom(FIXED_PARTITION_OFFSET(boot_partition), &bootloader_image_hdr,
```

Comment:
Expplanation: we don't have kernel ready yet, i.e., flash driver is still off. Use ROM code to retrieve flash data.